Formula/rex.rb

```diff
@@ -12,186 +12,224 @@ class Rex < Formula
 
   uses_from_macos "perl"
 
-  resource "Module::Build" do
-    # AWS::Signature4 requires Module::Build v0.4205 and above, while standard
-    # MacOS Perl installation has 0.4003
-    url "https://cpan.metacpan.org/authors/id/L/LE/LEONT/Module-Build-0.4229.tar.gz"
-    sha256 "1fe491a6cda914b01bc8e592faa2b5404e9f35915ca15322f8f2a8d8f9008c18"
-  end
+  if MacOS.version < :big_sur
+    resource "Module::Build" do
+      # AWS::Signature4 requires Module::Build v0.4205 and above, while standard
+      # MacOS Perl installation has 0.4003
+      url "https://cpan.metacpan.org/authors/id/L/LE/LEONT/Module-Build-0.4231.tar.gz"
+      sha256 "7e0f4c692c1740c1ac84ea14d7ea3d8bc798b2fb26c09877229e04f430b2b717"
+    end
 
-  resource "AWS::Signature4" do
-    url "https://cpan.metacpan.org/authors/id/L/LD/LDS/AWS-Signature4-1.02.tar.gz"
-    sha256 "20bbc16cb3454fe5e8cf34fe61f1a91fe26c3f17e449ff665fcbbb92ab443ebd"
-  end
+    resource "Clone::Choose" do
+      url "https://cpan.metacpan.org/authors/id/H/HE/HERMES/Clone-Choose-0.010.tar.gz"
+      sha256 "5623481f58cee8edb96cd202aad0df5622d427e5f748b253851dfd62e5123632"
+    end
 
-  resource "Clone::Choose" do
-    url "https://cpan.metacpan.org/authors/id/H/HE/HERMES/Clone-Choose-0.010.tar.gz"
-    sha256 "5623481f58cee8edb96cd202aad0df5622d427e5f748b253851dfd62e5123632"
-  end
+    resource "Exporter::Tiny" do
+      url "https://cpan.metacpan.org/authors/id/T/TO/TOBYINK/Exporter-Tiny-1.002002.tar.gz"
+      sha256 "00f0b95716b18157132c6c118ded8ba31392563d19e490433e9a65382e707101"
+    end
 
-  resource "Devel::Caller" do
-    url "https://cpan.metacpan.org/authors/id/R/RC/RCLAMP/Devel-Caller-2.06.tar.gz"
-    sha256 "6a73ae6a292834255b90da9409205425305fcfe994b148dcb6d2d6ef628db7df"
-  end
+    resource "JSON::MaybeXS" do
+      url "https://cpan.metacpan.org/authors/id/E/ET/ETHER/JSON-MaybeXS-1.004003.tar.gz"
+      sha256 "5bee3b17ff9dcffd6e99ab8cf7f35747650bfce1dc622e3ad10b85a194462fbf"
+    end
 
-  resource "Encode::Locale" do
-    url "https://cpan.metacpan.org/authors/id/G/GA/GAAS/Encode-Locale-1.05.tar.gz"
-    sha256 "176fa02771f542a4efb1dbc2a4c928e8f4391bf4078473bd6040d8f11adb0ec1"
-  end
+    resource "Scalar::List::Utils" do
```

Comment:
The way this is currently implemented, all the resources in this block get installed on Linux.  That's mostly desired, except for `Scalar::List::Utils`, which you mentioned is only need on macOS before Big Sur, but not Linux.  Is it okay to wrap put this inside an `on_macos` block inside of the `if MacOS.version` block?  I'm happy to organize this differently if there a better way.